pkg/secrethub/dir.go

```diff
@@ -2,13 +2,16 @@ package secrethub
 
 import (
 	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/internals/api/uuid"
 	"github.com/secrethub/secrethub-go/internals/errio"
 )
 
 // DirService handles operations on directories from SecretHub.
 type DirService interface {
 	// Create a directory at a given path.
 	Create(path string) (*api.Dir, error)
+	// Get returns the directory on the given path.
+	Get(id uuid.UUID) (*api.Dir, error)
```

Comment:
> Why do we have a UUID parameter here why all other parameters are path strings? I'm guessing it's because we need it somewhere in the CLI, but should that one use case dictate the signature of the client?

We'd like to move to IDs everywhere. This is the first move in that direction. Note that by referencing by ID, the argument is robust for changes of the path (for when we implement moving directories and secrets).

To be backwards compatible, do you think we should name the function `GetByID`? We can then also add e.g. `DeleteByID` and could add a `Get` or `GetByPath` if we'd like.

Reply:
> The comment says on the given path but it actually is a UUID.

Good catch :+1: , I've updated the comments.

Reply:
I think the `GetByID` is a pretty good middle road. In the future, I think we'll keep the paths as external signature only we may add ID arguments too. Internally things will change to no longer use the `blind_name` as HTTP parameter but use the `id` instead (e.g. by fetching and caching a tree of a repo), but externally we'll still support paths. 

So if you want to expose something by ID, then a specific `DoByID` function makes sense. We could also accept a string parameter that can be a path or a UUID, but I think that complicates things too much.